custom_components/opensprinkler/__init__.py

```diff
@@ -47,12 +51,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     url = entry.data.get(CONF_URL)
     password = entry.data.get(CONF_PASSWORD)
+    mqtt_option = entry.data.get(CONF_MQTT, DEFAULT_MQTT)
+
     try:
         controller = OpenSprinkler(url, password)
         controller.refresh_on_update = False
 
+        if mqtt_option:
```

Comment:
Based on the little I looked at from the other components we’ll probably need more stringent checks here to make sure the install actually has mqtt enabled.

Reply:
I thought about this a little and I feel like this is an "advanced" user option, so really its up to the user to make sure their setup is correct. Even if we check that the MQTT integration is enabled and check the OS config that MQTT is setup/enabled, there is no guarantee that it would work (even if we compare the two). Of course we should document this somewhere.

Which is why I am thinking it should be an option instead, which the user can easily enable/disable and switch between MQTT and the coordinator. We could even check for the firmware version as well as the OS MQTT config is enabled, otherwise disable/hide the option.

Reply:
Ok, I just noticed I think they were doing some pretty simple checks in those other integrations. If not yeah it’s advanced for sure so can probably leave as is.